axum-login/src/auth.rs

```diff
@@ -128,7 +128,12 @@ where
                     auth_cx.current_user = user;
 
                     request.extensions_mut().insert(auth_cx.clone());
-                    request.extensions_mut().insert(auth_cx.current_user);
+                    request
+                        .extensions_mut()
+                        .insert(auth_cx.current_user.clone());
```

Comment:
Is the idea in keeping this to preserve backwards compatibility?

I'm a little concerned is confusing to have `Option<User>` and `User` as extensions.

Reply:
Yes, lines 131-133 are for backwards compatibility while lines 134-136 implements the extension as described in the doc examples.

I am open to dropping lines 131-133 if you're not concerned with backwards compatibility. This change should be relatively safe as the `Option<User>` extension wasn't documented.

Reply:
I think it's safe to drop given we're pre-1.0.

Reply:
Sounds good, I have updated the pull request.